allennlp/training/trainer.py

```diff
@@ -389,7 +389,20 @@ def _data_parallel(self, batch):
         of torch.nn.parallel.data_parallel to support the allennlp model
         interface.
         """
+        metadata_batch_size = len(batch['metadata']) if 'metadata' in batch and isinstance(batch['metadata'],list) else None
```

Comment:
Please could you add a test for this functionality and annotate it as optional, depending on whether or not the environment has multiple GPUs - you can use 
```
@pytest.mark.skipif(torch.cuda.device_count() < 2,
                    reason="Need multiple GPUs.")
```

Reply:
I could definitely use some help with that, being a greenhorn! Given that the issue only surfaces in `forward()`, should I write a dummy model that verifies the correspondence between input tensors and metadata and a trainer config to run it? Is there boilerplate code somewhere in tests that I can use for this to avoid inventing the wheel?